src/python/include/Persistent_cohomology_interface.h

```diff
@@ -73,6 +73,60 @@ persistent_cohomology::Persistent_cohomology<FilteredComplex, persistent_cohomol
     return persistence;
   }
 
+  void top_dimensional_cofaces(std::vector<int> & cofaces, int splx){
+    if (stptr_->dimension(stptr_->simplex(splx)) == stptr_->dimension()){cofaces.push_back(stptr_->simplex(splx));}
+    else{  for (auto v : stptr_->coboundary_simplex_range(stptr_->simplex(splx))){top_dimensional_cofaces(cofaces, stptr_->key(v));}  }
```

Comment:
There is going to be a lot of redundancy here. For instance, in 3d, starting from a vertex, the 8 incident cubes will appear with multiplicity 6? I think you should take advantage of the filtration value. As soon as you find a coface (of codimension 1) with the same filtration value, use that one as base, it will lead to a top-dimensional cell with the same filtration value.

Reply:
Agreed. Thanks, I modified the code.